components/director/internal/domain/application/service.go

```diff
@@ -403,6 +413,10 @@ func (s *service) associateRelatedBundleInstanceAuthWithNewScenarios(ctx context
 		return err
 	}
 
+	if err = s.deleteDefaultEventing(ctx, tenantID, labelInput.ObjectID, scenariosToRemove); err != nil {
+		return err
+	}
```

Comment:
When you remove the application from formation with it's defaultEventing runtime the label from this runtime should be deleted. However new runtime should be assigned as defaultEventing, since the application can have more than one runtime in scenario with and therefore a new candidate for eventing.